### PR TITLE
grafana-ui: Add checkbox value field

### PR DESCRIPTION
**Pull request by AgnesToulet**
**What this PR does / why we need it**:
Update the checkbox component to be able to use the `value` field (that can't be used currently because the `value` field is used to fill the `checked` field). 

This would allow me to have multiple checkboxes updating an array field, see the example [here](https://codesandbox.io/s/usecontroller-checkboxes-99ld4?file=/src/App.js:460-613).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-ui/src/components/Forms/Checkbox.tsx [hunk 1]
```diff
@@ -12,10 +12,11 @@ export interface CheckboxProps extends Omit<HTMLProps<HTMLInputElement>, 'value'
   label?: string;
   description?: string;
   value?: boolean;
+  internalValue?: string | number;
```
Comment (AgnesToulet) [inline packages/grafana-ui/src/components/Forms/Checkbox.tsx:15]: Not sure about this name TBH but couldn't find anything else.

**Comment by Clarity-89** [comment]
> This would allow me to have multiple checkboxes updating an array field, see the example [here](https://codesandbox.io/s/usecontroller-checkboxes-99ld4?file=/src/App.js:460-613).

i think you might be able to use `name` attribute on checkboxes instead of value to achieve this.

File: packages/grafana-ui/src/components/Forms/Checkbox.tsx (hunk 1 above)
Comment (Clarity-89) [reply packages/grafana-ui/src/components/Forms/Checkbox.tsx:15]: Maybe smth like `htmlValue` to make it clear that it's an original HTML attribute (although not sure how clear that really is).

**Comment by Clarity-89** [comment]
> > This would allow me to have multiple checkboxes updating an array field, see the example [here](https://codesandbox.io/s/usecontroller-checkboxes-99ld4?file=/src/App.js:460-613).
> 
> i think you might be able to use `name` attribute on checkboxes instead of `value` to achieve this.

Looked into the Enterprise PR and it seems like `name` won't work here.

File: packages/grafana-ui/src/components/Forms/Checkbox.tsx (hunk 1 above)
Comment (Clarity-89) [reply packages/grafana-ui/src/components/Forms/Checkbox.tsx:15]: Also for a case like this I think we'd add a comment about what this prop is for.

**Review by Clarity-89 — APPROVED** [review]

**Review by Clarity-89 — APPROVED** [review]

**Pull request merged**